### PR TITLE
NServiceBus.Extensions.DependencyInjection documentation

### PR DESCRIPTION
**Pull request by timbussmann**
Adds a main page for the new package with doco snippets on how to configure both the native serviceprovider as well as third party containers using Autofac as an example here.

### CODE REVIEW

File: samples/dependency-injection/Extensions.DependencyInjection/Extensions.DependencyInjection_1/Sample/Program.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        var endpointConfiguration = new EndpointConfiguration("Samples.NServiceBus.Extensions.DependencyInjection");
+
+        // Use the default ServiceProvider factory, using Microsoft's built-in DI container:
+        var containerSettings = endpointConfiguration.UseContainer(new DefaultServiceProviderFactory());
+        containerSettings.ServiceCollection.AddSingleton(new MyService());
+
+        #endregion
+
+        endpointConfiguration.UsePersistence<LearningPersistence>();
```
Comment (andreasohlund) [inline samples/dependency-injection/Extensions.DependencyInjection/Extensions.DependencyInjection_1/Sample/Program.cs]: Nitpick: No need for storage

**Comment by timbussmann** [comment]
the build seems to fail because it can't find the package for the component, is there a way to make this work while this isn't released yet? @DavidBoike

**Comment by DavidBoike** [comment]
You can put your alpha on myget and never release to NuGet

File: nservicebus/dependency-injection/nservicebus-dependencyinjection.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+## Usage with ServiceCollection
+
+The following snippet shows how to configure NServiceBus to use Microsoft's built-in dependency injection container:
+
+snippet: usecontainer-servicecollection
+
+
+## Usage with third party containers
+
+NServiceBus can also be configured to work with any third party dependency injection which implements the `Microsoft.Extensions.DependencyInjection` abstraction. To use a third party dependency injection container, pass the specfic container's `IServiceProviderFactory` to the `UseContainer` configuration method. The following snippet shows this approach, using Autofac (`Autofac.Extensions.DependencyInjection`) as an example:
```
Comment (SeanFeldman) [inline nservicebus/dependency-injection/nservicebus-dependencyinjection.md]:
```suggestion
NServiceBus can also be configured to work with any third party dependency injection container which implements the `Microsoft.Extensions.DependencyInjection` abstraction. To use a third-party dependency injection container, pass the specific container's `IServiceProviderFactory` to the `UseContainer` configuration method. The following snippet shows this approach, using Autofac (`Autofac.Extensions.DependencyInjection`) as an example:
```

File: nservicebus/dependency-injection/nservicebus-dependencyinjection.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+
+## Configuring the container
+
+`UseContainer` provides a settings class which gives advanced configuration options:
+
+
+### IServiceCollection access
+
+Via the settings, the underlying `IServiceCollection` can be accessed to add additional service registrations.
```
Comment (SeanFeldman) [inline nservicebus/dependency-injection/nservicebus-dependencyinjection.md]:
```suggestion
The settings provide access to the underlying `IServiceCollection` that can be used to add additional service registrations.
```

File: nservicebus/dependency-injection/nservicebus-dependencyinjection.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+snippet: settings-servicecollection
+
+
+### ContainerBuilder
+
+Native container APIs can be accessed by the `ConfigureContainer` API.
```
Comment (SeanFeldman) [inline nservicebus/dependency-injection/nservicebus-dependencyinjection.md]:
```suggestion
Third-party container-native APIs can be accessed by with `ConfigureContainer`.
```

File: nservicebus/dependency-injection/nservicebus-dependencyinjection.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+snippet: settings-configurecontainer
+
+
+## Property Injection
+
+The `NServiceBus.Extensions.DependencyInjection` package does not support property injection out of the box. To enable property injection, please refer to the configured container's documentation.
```
Comment (SeanFeldman) [inline nservicebus/dependency-injection/nservicebus-dependencyinjection.md]:
```suggestion
The `NServiceBus.Extensions.DependencyInjection` package does **not** support property injection out of the box. To enable property injection, refer to the configured container's documentation.
```

**Comment by timbussmann** [comment]
> You can put your alpha on myget and never release to NuGet

but we have a package on myget and that seems to fail? Is there some mistake in the components related changes?

File: nservicebus/dependency-injection/nservicebus-dependencyinjection.md (hunk 2 above)
Comment (timbussmann) [reply nservicebus/dependency-injection/nservicebus-dependencyinjection.md]: I like those suggestions, but you're using third-party and third party, should align those cases?

**Comment by DavidBoike** [comment]
Building samples & snippets doesn't run unless content verification succeeds. Right now that's blocked by not having a /nservicebus/dependency-injection/extensions-dependencyinjection.md like your components.yaml entry says should exist.

The integrity tests are independent and are currently mad because your versioned snippets directory should have a(n empty) `prerelease.txt` file in it.

File: samples/dependency-injection/extensions-dependency-injection/Extensions.DependencyInjection_1/Sample/Sample.csproj [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <OutputType>Exe</OutputType>
+    <LangVersion>7.1</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus" Version="7.*" />
+    <PackageReference Include="NServiceBus.Extensions.DependencyInjection" Version="1.0.0-alpha0002" />
```
Comment (andreasohlund) [inline samples/dependency-injection/extensions-dependency-injection/Extensions.DependencyInjection_1/Sample/Sample.csproj]: So we need to make sure to change this once we push 1.0 right?

File: samples/dependency-injection/extensions-dependency-injection/sample.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
+---
+title: NServiceBus.Extensions.DependencyInjection Usage
+component: Extensions.DependencyInjection
+reviewed: 2019-09-10
```
Comment (andreasohlund) [inline samples/dependency-injection/extensions-dependency-injection/sample.md]:
```suggestion
reviewed: 2020-02-20
```

File: samples/dependency-injection/extensions-dependency-injection/Extensions.DependencyInjection_1/Sample/Sample.csproj (hunk 6 above)
Comment (timbussmann) [reply samples/dependency-injection/extensions-dependency-injection/Extensions.DependencyInjection_1/Sample/Sample.csproj]: yes

**Pull request merged**